utils/s2n_array.c

```diff
@@ -19,6 +19,17 @@
 
 #define S2N_INITIAL_ARRAY_SIZE 16
 
+extern bool s2n_array_is_valid(const struct s2n_array *array)
```

Comment:
I still prefer a `s2n_array_validate` function that returns a `S2N_RESULT` rather than `s2n_array_is_valid` that returns a `bool`. We would't have to chain checks with logic operators and we'd get better error messages than `post/pre condition violation`.

Then in all of the caller locations, you just `GUARD(s2n_array_validate(array))`.

Reply:
Thanks for changing it :)